### PR TITLE
Blog/starter pack

### LINKED ISSUE #5513: Blog post starter pack
Stakeholders: Internal users that want to create a new blog post
What?: A blog post starter pack with a template containing all the different options we currently offer 
Why?: We (CPT) are seeing an incline in requests for creating & publishing blogposts. The Marketing team provides [guidance](https://handbook.sourcegraph.com/departments/marketing/blog/) in the handbook on how to do this, but this process can be hard to follow for teammates that are not familiar with markdown or git. 

We have heard from our stakeholders that they would like this process to be easier and from our recent experience with assisting with these requests, we identified the following needs:

1. Create a starter/draft .md template that includes examples of what is included on the handbook page
- front matter example
- text options with markdown 
- image example
- video example
- Hubspot form example
- about the author section
3. Update instructions on the handbook page (we should notify and review changes with the Marketing team) (this can be done as a separate task)
4. Update instructions on the handbook page to encourage contributors to add an `About the author` section to blogposts.
5. Make the template package available somewhere (in the repo ideally) so that we can communicate it internally
As a company, we want to make it so that everyone can code and if we can make it easier for our teammates to create and publish their blog posts, we contribute to our company's mission

- [ ] #5553
- [ ] sourcegraph/handbook#4186
- [ ] #5541

### PR DESCRIPTION
**Pull request by zlonko**
This closes #5513. It creates an unpublished markdown file that includes all of the basic components we have abstracted for the blog.

### Changelog
- Adds `template.md`

### Test

1. Ensure prettier has standardized the proposed changes.

### CODE REVIEW

File: content/blogposts/blog-post-starter-pack.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+Add links with an anchor tag as you usually would: 
+
+<div className="text-center mb-5">
+  <a href="https://www.sourcegraph.com/search">Check out Sourcegraph Search!</a>
+</div>
+
+If you are wrapping an image in a link, please include a `title` on the anchor tag for accessibility.
+
+<a href="https://www.sourcegraph.com/search" title="Go to Sourcegraph Code Search">
```
Comment (st0nebraker) [inline content/blogposts/blog-post-starter-pack.md:238]: We should try to make the titles also describe the image along with the link since the screen reader will skip over the image when the code is written this way. It may take some trial and error to see what the screen reader says & what makes sense.

**Review by st0nebraker — CHANGES_REQUESTED** [review]
This is amazing! Thanks for writing such robust instructions, this will be so helpful for blog writers going forward.
Few comments below. Also, I'm seeing this blog publish at /blog/starter-pack, is there a way to make sure it won't publish on prod?

File: content/blogposts/blog-post-starter-pack.md (hunk 1 above)
Comment (zlonko) [reply content/blogposts/blog-post-starter-pack.md:238]: Great point! What do you think of adding this context to the description, then altering the link title and image alt text? Maybe something like this:

```suggestion
If you are wrapping an image in a link, please include a `title` on the anchor tag for accessibility, this way screen readers can pick up all the content. Screen readers may skip over the image, so it is important for the `title` to include a description of the image and intended action.

<a href="https://www.sourcegraph.com/search" title="An image of the Sourcegraph logo that links to Sourcegraph Code Search">
  <Figure 
    src="https://storage.googleapis.com/sourcegraph-assets/blog/default_hero_social.png"
    alt="Sourcegraph logo thumbnail"
  />
</a>
```

File: content/blogposts/blog-post-starter-pack.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+  portalId='2762526' 
+  formId='08e6c442-0e7c-4892-a262-76dae55ab497' 
+  targetId='#sign-up-form' />
+
+<div id="sign-up-form"></div>
+
+
+## Code
+
+Any code you write, you can style! The Blog uses Prism for code formatting. Specify the language at the front of the code block.
```
[the remark this answers is not in the document]
Comment (zlonko) [reply content/blogposts/blog-post-starter-pack.md:215]: Totally agree with you here. This is much better!

File: content/blogposts/blog-post-starter-pack.md (hunk 1 above)
Comment (st0nebraker) [reply content/blogposts/blog-post-starter-pack.md:238]: So the screen reader will read the link text & the image alt text only if the anchor tag surrounds a pure img tag, but when it surrounds the our Figure component it doesn't. Have you tested it? You can test with this blog `/blog/optimizing-a-code-intel-backend`. I think we should reconfigure the `Figure` component to make it more accessible, or encourage blog writers to not make complex images a link. Or instruct if the image is also a link, just to use the pure `` tag instead of Figure. What do you think?

Comment (bretthayes) [reply content/blogposts/blog-post-starter-pack.md:238]: I agree! I don't think we need to link Figures or images even @st0nebraker but maybe include a link the caption/cite if need be.

File: content/blogposts/blog-post-starter-pack.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+### HubSpot Form Example
+
+Use the `EmbeddedHubSpot` component to drop a custom HubSpot form into your post. In most cases, the `formId` is the only property that needs to be updated. This is the same `formId` dictated by the HubSpot console. Once added to your post, the HubSpot form will render wherever you place the `div` using the specified `id` value.
+
+<EmbeddedHubSpot 
+  portalId='2762526' 
+  formId='08e6c442-0e7c-4892-a262-76dae55ab497' 
+  targetId='#sign-up-form' />
+
+<div id="sign-up-form"></div>
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:210]: This API has changed and the component is now called `HubSpotForm`

```suggestion
### HubSpot Form Example

Use the `HubSpotForm` component to drop a custom HubSpot form into your post. In most cases, the `masterFormName` and `chiliPiper` props are the only properties that need to be updated. Once added to your post, the HubSpot form will render in its place. All the options for this component are:

- `formId` - an optional form ID if the form doesn't fall under one of our [master forms](https://docs.google.com/document/d/1qa393uAW_tjwNtH9jHf_d__PCJgmbsxNoxi9cO4XOcE/edit)
- `masterFormName` - an optional master form name; `contactMulti`, `contactEmail`, `gatedMulti`, or `gatedEmail` for general contact forms and gated forms that have multi fields or an email field
- `onFormSubmitted` - an optional callback function for triggering an action after a form is submitted
- `inlineMessage` - an optional message to display after a form is submitted. We have a [generic fallback](https://github.com/sourcegraph/about/blob/main/src/components/HubSpotForm.tsx#L288) if none is provided. 
- `chiliPiper` - an optional boolean (`true`/`false`) to enable Chili Piper. This is false by default.

<HubSpotForm
  masterFormName="contactEmail"
  chiliPiper={true}
/>
```

File: content/blogposts/blog-post-starter-pack.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (st0nebraker) [reply content/blogposts/blog-post-starter-pack.md]: @bretthayes , is there a way to make sure this mock blog post won't publish to prod?

File: content/blogposts/blog-post-starter-pack.md (hunk 3 above)
Comment (st0nebraker) [reply content/blogposts/blog-post-starter-pack.md]: Thank you!

File: content/blogposts/blog-post-starter-pack.md (hunk 1 above)
Comment (st0nebraker) [reply content/blogposts/blog-post-starter-pack.md]: Update made [8889b16](https://github.com/sourcegraph/about/pull/5553/commits/8889b16002511333223064eaba8e642c711bf206) :)

**Comment by st0nebraker** [comment]
This and [#4186](https://github.com/sourcegraph/handbook/pull/4186) are ready for review, @bretthayes ! :)

File: content/blogposts/blog-post-starter-pack.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+To set the thumbnails for this post, link to an image file using `socialImage` and `heroImage`. By default, blog posts use the Sourcegraph OG image. Using these properties in the frontmatter overrides the default. 
+
+- `socialImage` determines the image appearing on shareable links
+- `heroImage` determines the cover image appearing on `/blog`
+
+<hr/>
+
+## Components
+
+The Sourcegraph Blog is implemented in Next.js and MDX. One strength of this implementation is the ability to incorporate components in Markdown. The following components are available for blog posts:
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:73]:
```suggestion
The Sourcegraph Blog is implemented with Next.js with MDX support. One strength of this implementation is the ability to incorporate our custom components in Markdown. The following components are available for blog posts:
```

File: content/blogposts/blog-post-starter-pack.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+1. Alert
+2. Blockquote
+3. Figure
+4. TableWrapper
+5. Video
+6. YouTube
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:80]:
```suggestion
6. YouTube, and;
```

File: content/blogposts/blog-post-starter-pack.md [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+7. HubSpotForm
+
+Use these components to bring our blog post to life. Here are a few examples:
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:83]:
```suggestion
Use these components to bring your blog post to life. Here are a few examples:
```

File: content/blogposts/blog-post-starter-pack.md [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+        <td>0.2</td>
+      </tr>
+    </tbody>
+  </table>
+</TableWrapper>
+
+
+### YouTube
+
+You can easily embed YouTube videos using the `YouTube` component. The `id` value is found at the end of the URL for any YouTube video. To display the title below the video, add the `showTitle` property.
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:179]:
```suggestion
You can easily embed YouTube videos using the `YouTube` component. The two required properties are `title` and `id`.

A title should be descriptive for your YouTube video and is used for accessibility and the caption. It is also hidden by default. To optionally display the title below the video as a caption, add the `showTitle` property.

The `id` value is found at the end of the URL for any YouTube video.
```

File: content/blogposts/blog-post-starter-pack.md [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+<YouTube
+  title="Accelerate developer onboarding with Sourcegraph" 
+  id="DgwvhRW1Cbc"
+  showTitle={true}
+/>
+
+
+### Video
+
+The `Video` component supports .mp4 and .webm files. First, upload the two versions of the video to Google Cloud. Then, add the root names of these files to the `source` property, without including their extensions. You can choose to repeat a video by including the `loop` property, or display the caption using the `showCaption` property.
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:190]: The API will change slightly here in #5615

```suggestion
The required properties for the `Video` component are `mp4`, `webm`, `loop`, `caption`, and `title`. 

Since the component supports both `.mp4` and `.webm` files, you'll need to upload the two versions of the video to our Google Cloud [assets bucket](https://console.cloud.google.com/storage/browser/sourcegraph-assets/blog) under an appropriate sub directory. Then, add the file path from the Public URL of these files to the `mp4` and `webm` properties without their file extensions.

You can choose to repeat a video by including the `loop` property, optionally auto play the video by including the `autoPlay` property, or optionally display the caption using the `showCaption` property.
```

File: content/blogposts/blog-post-starter-pack.md [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+
+<Video 
+  source={{webm: 'blog/shift-left/01-code-reuse', mp4: 'blog/shift-left/01-code-reuse'}} 
+  loop={true}
+  title="Code reuse"
+  caption="An example of code search with Sourcegraph"
+  showCaption={true}
+/>
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:198]:
```suggestion
<Video 
  mp4="/blog/shift-left/01-code-reuse"
  webm="/blog/shift-left/01-code-reuse"
  loop={true}
  title="Code reuse"
  caption="An example of code search with Sourcegraph"
  showCaption={true}
/>
```

File: content/blogposts/blog-post-starter-pack.md [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+
+## Links
+
+Add links with an anchor tag as you usually would: 
+
+<div className="text-center mb-5">
+  <a href="https://www.sourcegraph.com/search">Check out Sourcegraph Search!</a>
+</div>
+
+Instead of linking images, please link the image's caption. This will allow our Figure components to be more accessible to screen readers. To add a link to an image, provide a `caption` and `link` property to the `Figure` component.
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:239]:
```suggestion
Instead of linking images using Markdown, please use the Figure component. This will allow our images to be more accessible to screen readers. To add a link to an image, provide a `caption` and `link` property to the `Figure` component.
```

File: content/blogposts/blog-post-starter-pack.md [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+<Figure 
+  src="https://storage.googleapis.com/sourcegraph-assets/blog/default_hero_social.png"
+  alt="Sourcegraph thumbnail"
+  caption="Check out Sourcegraph Search!"
+  link="https://www.sourcegraph.com/search"
+/>
+
+## About the author
+
+_Author blurb information goes here. You can chat with [author] on Twitter [@author-name](https://twitter.com/) or our community [Discord](https://discord.com/invite/vqsBW8m5Y8)._
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:250]:
```suggestion
## About the author

We encourage our authors to include an "about the author" section to their blog posts. Use the following Markdown for consistency throughout our blog posts.

_Author blurb information goes here. You can chat with [author] on Twitter [@author-name](https://twitter.com/) or our community [Discord](https://discord.com/invite/vqsBW8m5Y8)._
```

File: content/blogposts/blog-post-starter-pack.md [hunk 12]
```diff
@@ -0,0 +1,261 @@
...
+
+## TrySourcegraph CTA
+
+Note: A footer that includes the "Try Sourcegraph" CTA is automatically included at the end of each blog post.
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:254]:
```suggestion
Note: A footer that includes the "Try Sourcegraph" CTA is automatically included at the end of each blog post so there is no need to include it manually.
```

File: content/blogposts/blog-post-starter-pack.md [hunk 13]
```diff
@@ -0,0 +1,261 @@
...
+
+<hr/>
+
+## Need support?
+
+Please reach out to the Content Platform Team in #content-platform with any questions!
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:260]:
```suggestion
If you have any questions, concerns, or requests, please reach out to the Content Platform Team in #content-platform!
```

**Review by bretthayes — CHANGES_REQUESTED** [review]

**Comment by st0nebraker** [comment]
Adds some UI changes per Designs request. For point 5 below ([Figma](https://www.figma.com/file/o1QRtdQI0ozKq0n7ATrKlx/Marketing-DLS?node-id=3368%3A24494)), I'm holding off on adding a bg image to that CTA since it'll take some re-working of the `PostLayout` to span the whole screen & we'd like to get this in for the monthly update. Any thoughts and feelings around the bg image suggestion, @bretthayes ? 
image

File: content/blogposts/blog-post-starter-pack.md (hunk 9 above)
Comment (st0nebraker) [reply content/blogposts/blog-post-starter-pack.md:206]: Since this isn't in prod yet it's a breaking change right now. I'll bring it in once it's shipped!

**Review by bretthayes — CHANGES_REQUESTED** [review]

**Comment by bretthayes** [comment]
> Adds some UI changes per Designs request. For point 5 below ([Figma](https://www.figma.com/file/o1QRtdQI0ozKq0n7ATrKlx/Marketing-DLS?node-id=3368%3A24494)), I'm holding off on adding a bg image to that CTA since it'll take some re-working of the `PostLayout` to span the whole screen & we'd like to get this in for the monthly update. Any thoughts and feelings around the bg image suggestion, @bretthayes ? image

I would leave out the bg image for now and we can update the starter pack after #5615 is in, since it includes the new Background component.

File: content/blogposts/blog-post-starter-pack.md [hunk 14]
```diff
@@ -0,0 +1,271 @@
...
+
+You can choose to repeat a video by including the `loop` property, optionally auto play the video by including the `autoPlay` property, or optionally display the caption using the `showCaption` property.
+
+<Video 
+  source={{webm: 'blog/shift-left/01-code-reuse', mp4: 'blog/shift-left/01-code-reuse'}} 
+  loop={true}
+  title="Code reuse"
+  caption="An example of code search with Sourcegraph"
+  showCaption={true}
+/>
```
[the remark this answers is not in the document]
Comment (st0nebraker) [reply content/blogposts/blog-post-starter-pack.md:206]: This will cause the build to break though. When #5615 is merged I'll commit this. Or if this goes in first, #5615 can make this change to the file.

[the remark this answers is not in the document]
Comment (bretthayes) [reply content/blogposts/blog-post-starter-pack.md:197]: @st0nebraker Yes I know, that's what I'm saying - we should wait for #5615 to go in first since we don't want to encourage people to write this code since it will break.

[the remark this answers is not in the document]
Comment (st0nebraker) [reply content/blogposts/blog-post-starter-pack.md:197]: Can you break the bug fix into a separate PR to separate concerns? @bretthayes

[the remark this answers is not in the document]
Comment (bretthayes) [reply content/blogposts/blog-post-starter-pack.md:197]: @st0nebraker I updated the language slightly in the suggestion and reverted back to the original syntax since I confirmed that it's working in #5619

File: content/blogposts/blog-post-starter-pack.md [hunk 15]
```diff
@@ -0,0 +1,264 @@
+---
```
Comment (st0nebraker) [inline content/blogposts/blog-post-starter-pack.md:1]: Removing all these empty "Blogy McBlogerson" blogposts, no one knew about these blogposts. They probably outdate most of marketing and don't need to be in prod.

File: src/pages/blog/index.tsx [hunk 16]
```diff
@@ -35,11 +35,13 @@ export const getStaticProps: GetStaticProps = async ({ preview = false }) => {
             return { frontmatter: file.frontmatter, excerpt: content, slugPath: slug.slugPath }
         })
     )
-
+    // Don't show unpublished posts in index. Proper fix coming in: https://github.com/sourcegraph/about/issues/5620
+    const publishedPosts = posts.filter(post => post.frontmatter.published)
```
Comment (st0nebraker) [inline src/pages/blog/index.tsx:39]: As discussed, the short term fix to hide unpublished blogposts from the index. #5620 will be the more broad bugfix.

File: content/blogposts/blog-post-starter-pack.md [hunk 17]
```diff
@@ -0,0 +1,271 @@
...
+title: Blog Post Starter Pack
+description: This is a template for writing content on the Sourcegraph blog!
+authors:
+  - name: Sourcegraph
+    url: https://twitter.com/sourcegraph
+publishDate: 2013-01-01T18:00+02:00
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:7]: Ref: [official founding date](https://docs.google.com/document/d/19lePEDt9S6GODakIfzoB1HDf9p8ls1etwuFpXzAMLSo/edit)

```suggestion
publishDate: 2013-01-19T00:00
```

File: content/blogposts/blog-post-starter-pack.md [hunk 18]
```diff
@@ -0,0 +1,271 @@
...
+```yaml
+---
+title: Blog Post Starter Pack
+description: This is a template for writing content for the Sourcegraph blog!
+authors:
+  - name: The author name
+    url: https://example.com/
+  - name: Second authors name (optional)
+    url: https://example-2.com/
+publishDate: 2022-07-01T18:00+02:00
```
Comment (bretthayes) [inline content/blogposts/blog-post-starter-pack.md:41]: Just for kicks 🥚🙊

```suggestion
publishDate: 2013-01-19T00:00
```

**Review by bretthayes — CHANGES_REQUESTED** [review]
Should be good to go after these suggestions

**Review by st0nebraker — APPROVED** [review]
